### PR TITLE
Fixes #37261 Add gitlab ci configuration

### PR DESCRIPTION
**Pull request by dosas**

### CODE REVIEW

**Comment by dosas** [comment]
Build failing due to network issues:
```
/usr/share/ruby/net/protocol.rb:46:in `connect_nonblock': Unable to connect to https://forgeapi.puppet.com (for request /v3/releases/puppet-systemd-6.6.0): Connection reset by peer - SSL_connect (Faraday::ConnectionFailed)
```

File: .gitignore [hunk 1]
```diff
@@ -35,3 +35,5 @@ config/foreman.migrations/.applied
 spec/fixtures/katello-certs-check/certs/*.csr
 spec/fixtures/katello-certs-check/ca.key
 spec/fixtures/katello-certs-check/certs/*.srl
+
+.xml
```
Comment (ekohl) [inline .gitignore]: Line ending please

**Review by ekohl — CHANGES_REQUESTED** [review]
You missed the ` - ` between the issue number and the description in the commit message.

**Comment by dosas** [comment]
> You missed the `-` between the issue number and the description in the commit message.

The parser didn't mind ;)

**Review by ekohl — APPROVED** [review]
I think it's OK to include this. @evgeni / @ehelms is this also something we would use downstream? Could it conflict with anything?

**Comment by evgeni** [comment]
> I think it's OK to include this. @evgeni / @ehelms is this also something we would use downstream? Could it conflict with anything?

I recall talking to someone at cfgmgmtcamp that having gitlab ci stuff would be fine and we could *ideally* re-use it downstream -- or rip it out there.

**Comment by ehelms** [comment]
It could conflict, but I would expect that we can modify this or attempt to
find ways to co-exist since no one version of it should be the source of
truth given nothing verifies this directly in our repo.

**Comment by dosas** [comment]
How can we fix the stuck test?

**Comment by evgeni** [comment]
[test foreman-installer]

**Comment by dosas** [comment]
Should we merge or combine this with existing downstream configurations first?

**Comment by evgeni** [comment]
I'm fine merging this as-is and improving from there.

**Comment by dosas** [comment]
I would merge it, if I could ;)

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  variables:
+    PREFIX: "${RUBY_VERSION}"
+  script:
+    - apt update && apt install -y asciidoc-base docbook-xsl --no-install-recommends
+    - bundle exec rake install PREFIX="${PREFIX}" --trace
+    - bundle exec "${PREFIX}"/sbin/foreman-installer --help --scenario foreman --trace
+    - bundle exec "${PREFIX}"/sbin/foreman-installer --help --scenario foreman-proxy-content --trace
+    - bundle exec "${PREFIX}"/sbin/foreman-installer --help --scenario katello --trace
+    - bundle exec "${PREFIX}"/sbin/foreman-proxy-certs-generate --help --trace
+    - bundle exec "${PREFIX}"/sbin/foreman-proxy-certs-generate --help | grep -q certs-update-server
```
Comment (ekohl) [inline .gitlab-ci.yml]: I would like to somehow generalize these steps so we don't duplicate them between this and Jenkins. Perhaps not a limitation for this PR, but if you have any thoughts on a good way that would be appreciated.

Comment (dosas) [reply .gitlab-ci.yml]: That's a good idea. Generally there is way to much logic in the ci definitions.

* rake task (I am not really familiar with it but there is already a `bundle exec rake install`)
*  shell script
* Makefile

I will see if I can put it into rake.

Comment (dosas) [reply .gitlab-ci.yml]: @ekohl I added it to the rakefile

**Comment by dosas** [comment]
[test foreman-installer]

**Comment by dosas** [comment]
@evgeni Could you please trigger the installer-tests?

**Comment by evgeni** [comment]
ok to test

**Comment by dosas** [comment]
Is this good to merge?

**Review by ekohl — APPROVED** [review]

**Pull request merged**